src/middleware/validator.js

```diff
@@ -96,15 +96,17 @@ exports.validateInput = validationMap => async (ctx, next) => {
     logger.debug('Successfully validated user input')
   } catch (error) {
     ctx.status = 400
-    ctx.body = error
+    ctx.type = 'json'
+    ctx.body = JSON.stringify({ error: error.message })
     return logger.error(error)
   }
 
   try {
     performValidation(ctx, schema)
   } catch (error) {
     ctx.status = 400
-    ctx.body = error
+    ctx.type = 'json'
+    ctx.body = JSON.stringify({ error: error.message })
     return logger.error(error)
   }
```

Comment:
We could also refactor this function to make use of one try/catch as the error handling is the same. This could be done like below
```
try {
    const schema = createValidationSchema(validationMap)
    performValidation(ctx, schema)
    logger.info('Successfully validated user input')
  } catch (error) {
    ctx.status = 400
    ctx.type = 'json'
    ctx.body = JSON.stringify({ error: error.message })
    return logger.error(error.message)
  }
```

Reply:
Another change to this method is that the `Successfully validated user input` message is displayed after the validation has been performed and not after the schema has been built.

Reply:
Yes, I think this message is intended to write to log if an exception isn't thrown i.e. control flow can proceed to the actual mapping functionality. Also, it's a debug message (necessary?)

Reply:
I think either should be fine. I changed it from `debug` to `info` so we can see the request is being processed and completed the first middleware of validation. I think it will be useful to print out the completion of the middleware to indicate the progress of the request. What do you think?

Reply:
Sounds good